crates/net/downloaders/src/headers/reverse_headers.rs

```diff
@@ -557,6 +557,9 @@ where
                 Ordering::Equal => {
                     let OrderedHeadersResponse { headers, request, peer_id } =
                         PeekMut::pop(next_response);
+
+                    // shrink the buffer so that it doesn't grow indefinitely
+                    self.buffered_responses.shrink_to_fit();
```

Comment:
having this inside a loop is not ideal,

I think we can move it below the loop, since the error case would retrigger a new download